### ISSUE TITLE
Weird typing issue about `tuple` with some rare conditions

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#1056]
## Environment data

-   Language Server version: 2021.3.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.5(Anaconda)

## Steps to reproduce

- Run VSCode and open a workspace.
- Create or open a python source file **OUTSIDE** the workspace.
- Fill the source file with the sample code below.
- See a wrong warning based on wrong typing.

## Sample code

```python
l: "list[tuple[str, int]]" = [("", 0)]  # `tuple` must be used instead of `typing.Tuple`

for (s, i) in l:
    _ = s + ""  # `s` is typed as `str`, and it's OK here.
    _ = i + 0  # `i` is also typed as `str` instead of `int`, and here is a warning about it.
```

## Additional info

- The issue does **NOT** exist if the source file is placed inside the workspace.
- The issue does **NOT** exist if `typing.Tuple` is used instead of `tuple`.
- The issue does **NOT** exist if there is `from __future__ import annotations` ahead.

## Logs

```plain
[Info  - 下午3:37:18] Pylance language server 2021.3.1 (pyright 5d6db8d1) starting
[Info  - 下午3:37:18] Server root directory: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist
[Info  - 下午3:37:18] No configuration file found.
[Info  - 下午3:37:18] Setting pythonPath for service "a": "C:\Users\QWLHC\anaconda3\python.exe"
Search paths found for configured python interpreter:
  C:\Users\QWLHC\anaconda3\DLLs
  C:\Users\QWLHC\anaconda3\lib
  C:\Users\QWLHC\anaconda3
  C:\Users\QWLHC\anaconda3\lib\site-packages
  C:\Users\QWLHC\anaconda3\lib\site-packages\win32
  C:\Users\QWLHC\anaconda3\lib\site-packages\win32\lib
  C:\Users\QWLHC\anaconda3\lib\site-packages\Pythonwin
[Error - 下午3:37:18] stubPath d:\a\typings is not a valid directory.
[Info  - 下午3:37:19] Assuming Python version 3.8
[Info  - 下午3:37:19] Assuming Python platform Windows
[Info  - 下午3:37:19] Searching for source files
[Info  - 下午3:37:19] No source files found.
[Info  - 下午3:37:19] Background analysis(1) root directory: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist
[Info  - 下午3:37:19] Background analysis(1) started
Background analysis message: setConfigOptions
Background analysis message: setTrackedFiles
Background analysis message: markAllFilesDirty
Background analysis message: ensurePartialStubPackages
[FG] parsing: d:\a.py (63ms)
[FG] parsing: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 6ms] (309ms)
[FG] binding: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\builtins.pyi (80ms)
[FG] binding: d:\a.py (1ms)
Background analysis message: setFileOpened
Background analysis message: analyze
[BG(1)] analyzing: d:\a.py ...
[BG(1)]   parsing: d:\a.py (56ms)
[BG(1)]   parsing: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 5ms] (192ms)
[BG(1)]   binding: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\builtins.pyi (54ms)
[BG(1)]   binding: d:\a.py (1ms)
[BG(1)]   checking: d:\a.py ...
[BG(1)]     parsing: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 1ms] (25ms)
[BG(1)]     binding: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (3ms)
[BG(1)]     parsing: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\typing.pyi [fs read 2ms] (47ms)
[BG(1)]     binding: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\typing.pyi (20ms)
[BG(1)]     parsing: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stubs\typing-extensions\typing_extensions.pyi [fs read 3ms] (5ms)
[BG(1)]     binding: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stubs\typing-extensions\typing_extensions.pyi (2ms)
[BG(1)]     parsing: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\abc.pyi [fs read 1ms] (3ms)
[BG(1)]     binding: c:\Users\QWLHC\.vscode\extensions\ms-python.vscode-pylance-2021.3.1\dist\typeshed-fallback\stdlib\abc.pyi (1ms)
[BG(1)]   checking: d:\a.py (156ms)
[BG(1)] analyzing: d:\a.py (463ms)
Background analysis message: getSemanticTokens
Background analysis message: getSemanticTokens
Background analysis message: resumeAnalysis
```

**Comment by Azureblade3808** [comment]
![image](https://user-images.githubusercontent.com/15389744/111434702-cfdaad80-873a-11eb-94c7-7c92c5a69731.png)

**Comment by Azureblade3808** [comment]
![image](https://user-images.githubusercontent.com/15389744/111435050-44155100-873b-11eb-8c82-9678246a0576.png)

Here is a Python source file inside the workspace with very same content, and the issue is gone.

**Comment by erictraut** [comment]
The lower-cased types `list` and `tuple` are not subscriptable in versions of python prior to Python 3.9. Within your workspace, you probably have a Python 3.9 interpreter selected, so pylance is evaluating the code in that context.

If you add a `from __future__ import annotations` to the top of the file, it will work in both contexts.

`list` and `tuple` should also work fine in a quoted annotation in older versions of Python, and they currently don't. I've fixed that issue.

Thanks for reporting the problem.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.3.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202132-17-march-2021
